pkg/actors/placement/client.go

```diff
@@ -104,20 +117,24 @@ var noop = func() {}
 
 // disconnect from the current server without any additional operation.
 func (c *placementClient) disconnect() {
-	c.disconnectFn(noop)
+	c.disconnectFn(noop, true)
 }
 
 // disconnectFn disconnects from the current server providing a way to run a function inside the lock in case of new disconnection occurs.
 // the function will not be executed in case of the stream is already disconnected.
-func (c *placementClient) disconnectFn(insideLockFn func()) {
+func (c *placementClient) disconnectFn(insideLockFn func(), closeConnection bool) {
 	c.streamConnectedCond.L.Lock()
 	if !c.streamConnAlive {
 		c.streamConnectedCond.Broadcast()
 		c.streamConnectedCond.L.Unlock()
 		return
 	}
 
-	c.drain(c.clientStream, c.clientConn)
+	if closeConnection {
+		c.drain(c.clientStream, c.clientConn)
```

Comment:
I think we can get rid of the bool here and conditional, and do the nil check in the drain func like you already have

Reply:
Elena mentioned this is intentional to reuse the same client conn for grpc load balancing. this is fine then 👍🏻